File: frontend/src/services/communityService.js
import api from '../utils/axiosConfig';

const communityService = {
  createPost: async (content, mediaUrl) => {
    const response = await api.post('/api/community/post', { content, mediaUrl });
    return response.data;
  },

  getFeed: async () => {
    const response = await api.get('/api/community/feed');
    return response.data;
  },

  toggleLike: async (postId) => {
    const response = await api.post(`/api/community/post/${postId}/like`);
    return response.data;
  },

  addComment: async (postId, content) => {
    const response = await api.post(`/api/community/post/${postId}/comment`, { content });
    return response.data;
  },

  getComments: async (postId) => {
    const response = await api.get(`/api/community/post/${postId}/comments`);
    return response.data;
  }
};

export default communityService;

File: Edu/src/main/java/com/example/edu/service/CommunityService.java
package com.example.edu.service;

import com.example.edu.dto.response.CommunityCommentDto;
import com.example.edu.dto.response.CommunityPostDto;
import com.example.edu.entity.User;

import java.util.List;

public interface CommunityService {
    CommunityPostDto createPost(User author, String content, String mediaUrl);
    List<CommunityPostDto> getAllPosts(User currentUser);
    void toggleLike(User user, Long postId);
    
    CommunityCommentDto addComment(User author, Long postId, String content);
    List<CommunityCommentDto> getPostComments(Long postId);
}
